APE9.rst

```diff
@@ -0,0 +1,142 @@
+
+date-last-revised: 2017 November 10
+
+type: Standard Track
+
+status: Discussion
+
+
+Abstract
+--------
+
+The goal of a timeseries object in astropy is to provide a core set of
+functionality for representing data where time is the physical event list. This
+includes both continuous values binned in time or event list type data.
+
+The motivation for creation of this class is to support representation and
+analysis of astrophysical timeseries data, including support for the Units, Time
+and Coordinates packages in Astropy. This object should also be compatible with
+other packages in the wider scientific Python ecosystem, and try to minimise
+overlap in functionality with these packages.
+
+This document proposes an implementation based on Table, with extra constraints
+and specific functionality dedicated to timeseries type data.
+
+Detailed description
+--------------------
+
+Many different areas of astrophysics have to deal with 1D timeseries data, i.e.
+either sampling a continuous variable at fixed times or counting some events
+binned into time windows. These types of applications require some basic
+functionality:
+
+#. Extending timeseries with extra rows
+#. Concatenating multiple timeseries objects
+#. Sorting
+#. Slicing / selecting time ranges (indexing)
+#. Re-binning and re-sampling timeseries
+#. Interpolating to different time stamps.
+#. Masking
+#. Support for subtraction and addition (e.g. background).
+#. Plotting and visualisation.
+#. Have high performance into \~millions of rows.
+
+While this functionality is found in non-domain specific packages such as
+pandas, a timeseries object in Astropy would also provide some functionality
+which is more related to Astronomy:
+
+#. Converting between time systems
+#. Astropy unit support
+#. Support variable width time bins.
+
+It is proposed that this functionality be implemented via a subclass of the
+``QTable`` class, with a few requirements specific to the timeseries class:
+
+#. A 'Time' index column exists which enforces unique indexes.
+#. The table is always sorted in terms of increasing time.
+
+
+Binned Data vs Sampled Data
+###########################
+
+There are two different types of data that a timeseries class or classes would
+have to support, sampled data and binned data. Sampled data is taken at a
+precise time stamp whereas binned data is a number of counts contained within a
+some time window. These two different types of timeseries data require different
+handling of methods such as re-binning or re-sampling, as well as for binned
+data the width of the bins must be stored.
+
+
+To distinguish between binned and sampled timseries we propose having two ways
+of constructing a timeseries, ``BinnedTimeSeries`` and ``SampledTimeSeries``.
+This distinction in the name of the classes will allow a easier to understand
+API. Some examples of constructing these objects is given below.
+
+
+Initialize a ``SampledTimeSeries`` with a time and a data column::
```

Comment:
This section should also demonstrate constructing these objects with Time and TimeDelta so that it's clear that's a thing.